gtsam/hybrid/HybridGaussianConditional.cpp

```diff
@@ -288,28 +310,39 @@ HybridGaussianConditional::shared_ptr HybridGaussianConditional::prune(
 
   // Check the max value for every combination of our keys.
   // If the max value is 0.0, we can prune the corresponding conditional.
-  auto pruner = [&](const Assignment<Key> &choices,
-                    const GaussianConditional::shared_ptr &conditional)
-      -> GaussianConditional::shared_ptr {
-    return (max->evaluate(choices) == 0.0) ? nullptr : conditional;
+  auto pruner =
+      [&](const Assignment<Key> &choices,
+          const GaussianFactorValuePair &pair) -> GaussianFactorValuePair {
+    if (max->evaluate(choices) == 0.0)
+      return {nullptr, std::numeric_limits<double>::infinity()};
+    else
+      return pair;
   };
 
-  auto pruned_conditionals = conditionals_.apply(pruner);
-  return std::make_shared<HybridGaussianConditional>(discreteKeys(),
-                                                     pruned_conditionals);
+  FactorValuePairs prunedConditionals = factors().apply(pruner);
+  return std::shared_ptr<HybridGaussianConditional>(
```

Comment:
Why not `make_shared` over here? The constructor is being called in both cases.

Reply:
Done